.circleci/test-example-repo.sh

```diff
@@ -4,8 +4,25 @@ set -ex
 
 TERMINUS_SITE=build-tools-$CIRCLE_BUILD_NUM
 
-BRANCH=$(echo $CIRCLE_BRANCH | grep -v '^\(master\|[0-9]\+.x\)$')
-PR_BRANCH=${BRANCH:+dev-$BRANCH}
+# If we are on the master branch
+if [[ $CIRCLE_BRANCH == "master" ]]
+then
+    PR_BRANCH="dev-master"
+else
+    # If this is a pull request use the PR number
+    if [[ -z "$CIRCLE_PULL_REQUEST" ]]
+    then
+        # Stash PR number
+        PR_NUMBER=${CIRCLE_PULL_REQUEST##*/}
```

Comment:
I'm not sure what is causing the discrepancy here.

As for the result, ultimately BUILD_TOOLS_VERSION is used to select a release via Composer, so it should be `dev-foo` for a PR on the `foo` branch, or `dev-master` for the master branch, or `^1` or `~1` or `1.x` for the 1.x branch.